src/book/02-system/01-foundations/04-nuclear.mdx

```diff
@@ -186,6 +186,20 @@ NBS files are NUClear Binary Stream files. To make it easier to serialise stream
 
 All values within this format are little endian.
 
+An `nbz` or `nbs.gz` file is a compressed `nbs` file using gzip.
+
+NBS files can have an acsossiated index file. This allows for quicker random access to the coresponding nbs file. An `nbs.idx` file has the folowwing frames repeated continuously, which are then compressed with gzip:
+
+| Name      | Type   | Description                                                                      |
+| --------- | ------ | -------------------------------------------------------------------------------- |
+| hash      | uint64 | the 64bit hash for the payload type                                              |
+| subtype   | uint32 | the id field of the message if it exists, otherwise 0                            |
+| timestamp | uint64 | timestamp field of the message if it exists or the emit timestamp in nanoseconds |
```

Comment:
```suggestion
| timestamp | uint64 | The `timestamp` of the message if it exists or the emit timestamp in nanoseconds |
```
What's the emit timestamp relative to? (Although I don't know if this is the right spot to talk about that?)